src/SnapshotFactory.php

```diff
@@ -25,20 +26,26 @@ public function __construct(DbDumperFactory $dumperFactory, Factory $filesystemF
         $this->filesystemFactory = $filesystemFactory;
     }
 
-    public function create(string $snapshotName, string $diskName, string $connectionName): Snapshot
+    public function create(string $snapshotName, string $diskName, string $connectionName, bool $compress = null): Snapshot
     {
         $disk = $this->getDisk($diskName);
 
         $fileName = $snapshotName.'.sql';
         $fileName = pathinfo($fileName, PATHINFO_BASENAME);
 
+        $compress = ($compress === true) ?: (bool) config('db-snapshots.compress', false);
```

Comment:
Why do we read the config file here? Can't we check the config file in the code that call  this `create` method and pass it as the last parameter?

Reply:
Yes, the idea was originally to let direct calls to this method (not through the `Spatie\DbSnapshots\Commands\Create`) to also have the benefit of configuration to determine what the default is.

Moving the config call to the caller (in this library, only `Spatie\DbSnapshots\Commands\Create` calls this method), means that only the console command will respect the `db-snapshots.compress` option.

Either way, it is a sensible request, I am making this change now.